sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/README.md

```diff
@@ -8,112 +8,211 @@ The authentication events trigger for Azure Functions allows you to implement a
+After creating the project, you'll need to install the required [NuGet package](https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents)
 
-When running your function in production, it is **highly recommended** to use the [Azure Functions Microsoft Entra ID authentication integration](https://learn.microsoft.com/azure/app-service/configure-authentication-provider-aad#-option-2-use-an-existing-registration-created-separately) for validating incoming tokens. Set the following function [application settings](https://learn.microsoft.com/azure/azure-functions/functions-how-to-use-azure-function-app-settings?tabs=portal#settings).
+```dotnetcli
+dotnet add package Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
+```
 
-1. Go to the "Authentication" tab in your Function App
-2. Click on "Add identity provider"
-3. Select "Microsoft" as the identity provider
-4. Select "Provide the details of an existing app registration"
-5. Enter the `Application ID` of the app that represents your API in Microsoft Entra ID
+---
 
-The issuer and allowed audience depends on the [`accessTokenAcceptedVersion`](https://learn.microsoft.com/azure/active-directory/develop/access-tokens) property of your application (can be found in the "Manifest" of the application).
+### Add the snippet
 
-If the `accessTokenAcceptedVersion` property is set to `2`:
-6. Set the `Issuer URL to "https://login.microsoftonline.com/{tenantId}/v2.0"
-7. Set an 'Allowed Audience' to the Application ID (`appId`)
+The function API is the source of extra claims for your token. For the purposes of this article, we're hardcoding the values for the sample app. In production, you can fetch information about the user from external data store.
 
-If the `accessTokenAcceptedVersion` property is set to `1` or `null`:
-6. Set the `Issuer URL to "https://sts.windows.net/{tenantId}/"
-7. Set an 'Allowed Audience' to the Application ID URI (also known as`identifierUri`). It should be in the format of`api://{azureFunctionAppName}.azurewebsites.net/{resourceApiAppId}` or `api://{FunctionAppFullyQualifiedDomainName}/{resourceApiAppId}` if using a [custom domain name](https://learn.microsoft.com/azure/dns/dns-custom-domain#:~:text=Azure%20Function%20App%201%20Navigate%20to%20Function%20App,Custom%20domain%20text%20field%20and%20select%20Validate.%20).
+In your trigger class, add the contents of the following snippet:
 
-By default, the Authentication event trigger will validate that Azure Function authentication integration is configured and it will check that the **client** in the token is set to `99045fe1-7639-4a75-9d4a-577b6ca3810f` (via the `azp` or `appid` claims in the token).
+```C# Snippet:AuthEventsTriggerExample
+try
+{
+    // Checks if the request is successful and did the token validation pass
+    if (request.RequestStatus == WebJobsAuthenticationEventsRequestStatusType.Successful)
```

Comment:
I have no idea where `request` comes from.   I can't be sure that I can copy/paste this snippet and use it.  There's context missing.

Take an Event Hubs example:

```csharp
[FunctionName("BindingToProducerClient")]
public static async Task Run(
    [TimerTrigger("0 */5 * * * *")] TimerInfo myTimer,
    [EventHub("<event_hub_name>", Connection = "<connection_name>")] EventHubProducerClient eventHubProducerClient)
{
    // IAsyncCollector allows sending multiple events in a single function invocation
    await eventHubProducerClient.SendAsync(new[]
    {
        new EventData($"Event 1 added at: {DateTime.Now}"),
        new EventData($"Event 2 added at: {DateTime.Now}")
    });
}
```

I can clearly see how to write my Function signature and I'm not using anything in the snippet that isn't defined here.   I can clearly see that `eventHubProducerClient` is a parameter to the Function itself.

Reply:
Added the full `run` function to make it clear as the examples per the other readmes